src/ripple/app/misc/impl/ValidatorList.cpp

```diff
@@ -0,0 +1,269 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of rippled: https://github.com/ripple/rippled
+    Copyright (c) 2015 Ripple Labs Inc.
+
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#include <BeastConfig.h>
+#include <ripple/app/misc/ValidatorList.h>
+#include <ripple/basics/Slice.h>
+#include <ripple/basics/StringUtilities.h>
+#include <ripple/core/ConfigSections.h>
+#include <boost/regex.hpp>
+#include <fstream>
+
+namespace ripple {
+
+static
+PublicKey
+asPublicKey(RippleAddress const& raPublicKey)
+{
+    auto const& blob = raPublicKey.getNodePublic();
+
+    if (blob.empty())
+        LogicError ("Can't convert invalid RippleAddress to PublicKey");
+
+    return PublicKey(Slice(blob.data(), blob.size()));
+}
+
+ValidatorList::ValidatorList (beast::Journal j)
+    : j_ (j)
+{
+}
+
+boost::optional<std::string>
+ValidatorList::member (RippleAddress const& identity) const
+{
+    std::lock_guard <std::mutex> sl (mutex_);
+
+    auto const publicKey = asPublicKey (identity);
+
+    auto ret = ephemeral_.find (publicKey);
+
+    if (ret != ephemeral_.end())
+        return ret->second;
+
+    ret = permanent_.find (publicKey);
+
+    if (ret != permanent_.end())
+        return ret->second;
+
+    return boost::none;
+}
+
+bool
+ValidatorList::trusted (RippleAddress const& identity) const
+{
+    return static_cast<bool> (member(identity));
```

Comment:
Instead of using `static_cast`, which calls `boost::optional`'s explicit `operator bool`, you could `return member(identity).is_initialized();`. With compiler optimizations, I think the only difference is stylistic.